crates/yulgen/src/operations/abi.rs

```diff
@@ -88,3 +116,36 @@ pub fn unpack(
 ) -> yul::Statement {
     statement! { abi_unpack([ptr], [array_size], [inner_data_size]) }
 }
+
+/// Reverts if the value is not left padded with the given number of bits.
+pub fn check_left_padding(size_bits: yul::Expression, val: yul::Expression) -> yul::Statement {
+    statement! {
+        if (iszero((is_left_padded([size_bits], [val])))) {
+            (revert(0, 0))
```

Comment:
Just re-stating what I commented above on the test code. I think we should consider introducing our own panic code and then do a `revert_with_panic(<some-code>)`. Basically I think the only time where we should revert with zero data is when someone actually uses `revert` with no data. In all other circumstances we should revert with a panic code and we should go with whatever panic code Solidity uses and if they don't use one we make up our own.